tests/pytests/functional/states/test_file.py

```diff
@@ -53,7 +54,13 @@ def do_GET(self):
 
         # Return the Etag header if we have the checksum
         if checksum:
-            self.send_header("Etag", checksum)
+            # IMPORTANT: This introduces randomness into the tests. The Etag header key
+            # will be converted to lowercase in the code... but if someone breaks that,
+            # it'll rear it's head here as random failures that are hard to reproduce.
+            # Any alternatives seem overly complex. So... don't break the case insensitivity
+            # in the code.
+            possible_etags = ["Etag", "ETag"]
```

Comment:
(Hilarious) **suggestion(non-blocking, entirely for the lulz)**: eTaG :trollface:

Reply:
More seriously, `etag` is certainly likely in the wild.

Also, can we cut the comments insinuating that this is some sort of workaround, when it's what every system that receives HTTP responses must do by default?

I am surprised that the HTTP client isn't normalising header names itself already, or providing some sort of case-insensitive map. That's what e.g. `requests` does.